six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -304,8 +305,14 @@ void NITFWriteControl::initialize(mem::SharedPtr<Container> container)
                 subheader.getImageCompression().set("C8");
 
                 //calculate comrat
-                int comratInt = (int)((j2kCompression * nbpp * 10.0) + 0.5);
-                std::string comrat = FmtX("N%03d", comratInt);
+                const int comratInt = (int)((j2kCompression * nbpp * 10.0) + 0.5);
+                const bool isNumericallyLossless = (bool)mOptions.getParameter(
+                        OPT_J2K_COMPRESSION_LOSSLESS, Parameter(false));
+                // We are assuming the image is not compressed so heavily as
+                // to be visually lossy
+                const char comratChar = isNumericallyLossless ? 'N' : 'V';
+
+                const std::string comrat = FmtX("%c%03d", comratChar, comratInt);
```

Comment:
Avoid using `FmtX()` (under the hood it uses a fixed-length char array.  Use `std::ostringstream` with `std::setw()`.